lib/connect.ts

```diff
@@ -65,1354 +69,648 @@ import utils from "./utils";
+    getOrders(): Promise<AxiosResponse<any>> {
+        return this._get('orders', null, null, formatResponse);
     };
 
-    /**
-     * Get list of order history.
-     * @method getOrderHistory
-     * @memberOf KiteConnect
-     * @instance
-     * @param {string} order_id ID of the order whose order details to be retrieved.
-     */
-    self.getOrderHistory = function(order_id) {
-        return  _get("order.info", {"order_id": order_id}, null, formatResponse);
+    getOrderHistory(order_id: string | number): Promise<AxiosResponse<any>> {
+        return this._get('order.info', { 'order_id': order_id }, null, formatResponse);
     };
 
-    /**
-     * Retrieve the list of trades executed.
-     * @method getTrades
-     * @memberOf KiteConnect
-     * @instance
-     */
-    self.getTrades = function() {
-        return _get("trades", null, null, formatResponse);
+    getTrades(): Promise<AxiosResponse<any>> {
+        return this._get('trades', null, null, formatResponse);
     };
 
-    /**
-     * Retrieve the list of trades a particular order).
-     * An order can be executed in tranches based on market conditions.
-     * These trades are individually recorded under an order.
-     * @method getOrderTrades
-     * @memberOf KiteConnect
-     * @instance
-     * @param {string} order_id ID of the order whose trades are to be retrieved.
-     */
-    self.getOrderTrades = function(order_id) {
-        return  _get("order.trades", {"order_id": order_id}, null, formatResponse);
+    getOrderTrades(order_id: string | number): Promise<AxiosResponse<any>> {
+        return this._get('order.trades', { 'order_id': order_id }, null, formatResponse);
     };
 
-    /**
-     * Fetch required margin for order/list of orders
-     * @method orderMargins
-     * @memberOf KiteConnect
-     * @instance
-     * @param {Object[]} orders   Margin fetch orders.
-     * @param {string} orders[].exchange    Name of the exchange(eg. NSE, BSE, NFO, CDS, MCX)
-     * @param {string} orders[].tradingsymbol   Trading symbol of the instrument
-     * @param {string} orders[].transaction_type   eg. BUY, SELL
-     * @param {string} orders[].variety   Order variety (regular, amo, bo, co etc.)
-     * @param {string} orders[].product   Margin product to use for the order
-     * @param {string} orders[].order_type   Order type (MARKET, LIMIT etc.)
-     * @param {number} orders[].quantity   Quantity of the order
-     * @param {number} orders[].price   Price at which the order is going to be placed (LIMIT orders)
-     * @param {number} orders[].trigger_price   Trigger price (for SL, SL-M, CO orders)
-     * @param {string} mode    (optional) Compact mode will only give the total margins
-     */
-    self.orderMargins = function(orders, mode=null) {
-        return _post("order.margins", orders, null, null, true,
-                    {"mode":mode});
+    orderMargins(orders: Order[], mode = null): Promise<AxiosResponse<any>> {
```

Comment:
Missing types for orders and mode. Maybe use: `orderMargins(orders:Record<string, any>, mode:string)`